sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala

```diff
@@ -109,7 +111,14 @@ trait GpuMapInBatchExec extends ShimUnaryExecNode with GpuPythonExecBase {
           numOutputRows += cb.numRows
           cb
         }
-    } // end of mapPartitionsInternal
+    } // end of func
+
+    if (isBarrier) {
+      child.executeColumnar().barrier().mapPartitions(func)
```

Comment:
Any special reason to use `mapPartitions` instead of `mapPartitionsInternal` here ?

Reply:
`barrier()` will get RDDBarrier() which doesn't have mapPartitionInternal